### DISCUSSION TITLE
Location of meeting changing to wrong town and zip code

### DISCUSSION THREAD

**Discussion by Area48Webmaster** [code4recovery/12-step-meeting-list#893] [Q&A]
Hi All, thank you for your continued and awesome service.

We have two meetings (at the same location) that are listed correctly in our database (airtable) but when importing the JSON the location of the meeting, specifically the town and zip code are changed to an incorrect town and zip code. I have tried manually changing the address to the correct one within the TSML plugin, right on wordpress, but before I can hit the update button it changes back to the wrong town and zip code. Thank you in advance for any help you can provide.

Some specifics: 
Our website - [nenyaa.org](url)
Meetings affected - https://nenyaa.org/locations/jefferson-methodist-church/?tsml-day=any&tsml-query=Covered+Bridge+Group
Address should be: 158 Main St., Jefferson, NY 12093

**Comment by tim-rohrer** [comment]
Hello!

I entered the address into "https://maps.google.com" and it returned showing in Worcester, NY. It appears that is the source of the problem. 

Two things you can do:
1. The Google maps page provides a link to suggest an edit. Google may then fix this long term.
2. In the meantime, you should be able to use a child theme and local `functions.php`, or other method of your choosing, to  override the address TSML. We provide [general directions](https://github.com/code4recovery/12-step-meeting-list/wiki/Frequently-Asked-Questions#i-need-to-correct-a-meeting-address-or-change-a-pins-location) on how to do this, but we have an error in our code sample. Try this:

```php
	if (function_exists('tsml_custom_addresses')) {
		tsml_custom_addresses(array(
			'158 Main St., Jefferson, NY 12093' => array(
				'formatted_address' => '158 Main St., Jefferson, NY 12093, USA',
				'city' => 'Jefferson',
				'latitude' => 48.858372, // Enter whatever is the correct latitude
				'longitude' => 2.294481, // Enter whatever is the correct longitude
			),
		));
	}
```

**Comment by Area48Webmaster** [comment]
Thank you. I will give the override a try. Is there a specific place in
functions.php that this code should be added?

On Mon, Aug 1, 2022 at 9:25 AM Tim Rohrer ***@***.***> wrote:

> Hello!
>
> I entered the address into "https://maps.google.com" and it returned
> showing in Worcester, NY. It appears that is the source of the problem.
>
> Two things you can do:
>
>    1. The Google maps page provides a link to suggest an edit. Google may
>    then fix this long term.
>    2. In the meantime, you should be able to use an override in TSML to
>    correct the address. We provide general directions
>    <https://github.com/code4recovery/12-step-meeting-list/wiki/Frequently-Asked-Questions#i-need-to-correct-a-meeting-address-or-change-a-pins-location>
>    on how to do this, but we have an error in our code sample. Try this:
>
> 	if (function_exists('tsml_custom_addresses')) {
> 		tsml_custom_addresses(array(
> 			'158 Main St., Jefferson, NY 12093' => array(
> 				'formatted_address' => '158 Main St., Jefferson, NY 12093, USA',
> 				'city' => 'Jefferson',
> 				'latitude' => 48.858372, // Enter whatever is the correct latitude
> 				'longitude' => 2.294481, // Enter whatever is the correct longitude
> 			),
> 		));
> 	}
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/893#discussioncomment-3297109>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASKFRH5ROQPBW3XP444RNFLVW7F3RANCNFSM55E4JFRQ>
> .
> You are receiving this because you authored the thread.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/893/comments/3297109@
> github.com>
>


-- 
*Matt E.*
Area 48 Technology Chair

**Reply by tim-rohrer** [reply]
In general, you should create a child theme or use some other method to add code to a `functions.php`. Don't edit the plugin's `functions.php` directly because that will be overwritten when the next update is published.

As far as Google and the address:

<img width="405" alt="image" src="https://user-images.githubusercontent.com/7756865/183784937-4715397d-9b55-4127-a241-68aab8814189.png">

**Comment by Area48Webmaster** [comment]
Also, when I type 158 Main Street Jefferson NY into google maps I am not
taken to the Worcester NY location but instead to the correct address that
we want the meeting finder to show. Can you clarify what you entered into
google maps to be taken to Worcester? Thanks again!

On Mon, Aug 8, 2022 at 5:43 PM HMB Webmaster ***@***.***> wrote:

> Thank you. I will give the override a try. Is there a specific place in
> functions.php that this code should be added?
>
> On Mon, Aug 1, 2022 at 9:25 AM Tim Rohrer ***@***.***>
> wrote:
>
>> Hello!
>>
>> I entered the address into "https://maps.google.com" and it returned
>> showing in Worcester, NY. It appears that is the source of the problem.
>>
>> Two things you can do:
>>
>>    1. The Google maps page provides a link to suggest an edit. Google
>>    may then fix this long term.
>>    2. In the meantime, you should be able to use an override in TSML to
>>    correct the address. We provide general directions
>>    <https://github.com/code4recovery/12-step-meeting-list/wiki/Frequently-Asked-Questions#i-need-to-correct-a-meeting-address-or-change-a-pins-location>
>>    on how to do this, but we have an error in our code sample. Try this:
>>
>> 	if (function_exists('tsml_custom_addresses')) {
>> 		tsml_custom_addresses(array(
>> 			'158 Main St., Jefferson, NY 12093' => array(
>> 				'formatted_address' => '158 Main St., Jefferson, NY 12093, USA',
>> 				'city' => 'Jefferson',
>> 				'latitude' => 48.858372, // Enter whatever is the correct latitude
>> 				'longitude' => 2.294481, // Enter whatever is the correct longitude
>> 			),
>> 		));
>> 	}
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/code4recovery/12-step-meeting-list/discussions/893#discussioncomment-3297109>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ASKFRH5ROQPBW3XP444RNFLVW7F3RANCNFSM55E4JFRQ>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> <code4recovery/12-step-meeting-list/repo-discussions/893/comments/3297109
>> @github.com>
>>
>
>
> --
> *Matt E.*
> Area 48 Technology Chair
>


-- 
*Matt E.*
Area 48 Technology Chair

**Reply by brianw-area24** [reply]
When I put "158 Main St, Jefferson, NY" into Google maps, I get what appears to be the correct place. However, if I add the zip code "158 Main St, Jefferson, NY 12093", I get Worcester back.
Very Strange.

**Comment by Area48Webmaster** [comment]
Hi All, Google has not made any changes to this issue on their end so I
would like to do the override in TSML. I will admit, I have not done any
edits to coding like this before, nor have I created a child theme before.
Would you be able to provide a step by step for me? Please let me know if
you need any details from me. Thank you in advance.

On Thu, Aug 11, 2022 at 1:31 AM brianw-area24 ***@***.***>
wrote:

> When I put "158 Main St, Jefferson, NY" into Google maps, I get what
> appears to be the correct place. However, if I add the zip code "158 Main
> St, Jefferson, NY 12093", I get Worcester back.
> Very Strange.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/893#discussioncomment-3371728>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASKFRHYTUFT6P7H7NEK4J3DVYSF2RANCNFSM55E4JFRQ>
> .
> You are receiving this because you authored the thread.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/893/comments/3371728@
> github.com>
>


-- 
*Matt E.*
Area 48 Technology Chair

**Comment by cassandradc** [comment]
I think we are experiencing a similar problem. We have a meeting address listed in Meeting Guide correctly, but when users use the Get Direction and Google Maps is opened, it changes the street number.

We are on Word Press. I think I need to make a request to Google to correct, but I'm not sure what even to ask them.

Meeting Listing: https://aa-dc.org/locations/central-group-location-2?tsml-day=2&tsml-query=central
Listed Address: 5121 Georgia Ave NW, Washington, DC 20011

When "Get Directions" is selected, it opens a new tab and autopopulates Google Maps with the wrong street number (2125 Georgia Ave): https://www.google.com/maps/dir/?api=1&destination=38.952642,-77.0272327

**Reply by gkovats** [reply]
Looking into the best way to override the incorrect longitude. If you can try using the solution above to override the incorrect data with your own:
```php
	if (function_exists('tsml_custom_addresses')) {
		tsml_custom_addresses(array(
			'5121 Georgia Ave NW, Washington, DC 20011' => array(
				'formatted_address' => '5121 Georgia Ave NW, Washington, DC 20011, USA',
				'city' => 'Washington', // ? what IS the city in DC?
				'latitude' => 38.952642,
				'longitude' => -77.0298076, // this appears to be the correct value
			),
		));
	}
```
